consensus/src/chained_bft/persistent_storage.rs

```diff
@@ -123,7 +123,10 @@ impl<T: Payload> RecoveryData<T> {
     }
 
     pub fn last_vote(&self) -> Option<Vote> {
-        self.last_vote.clone()
+        match &self.last_vote {
+            Some(v) if v.epoch() == self.epoch => Some(v.clone()),
+            _ => None,
+        }
```

Comment:
would prefer we prune the persistent store, what are your thoughts?

Reply:
yeah, I prefer to delete too, but I just didn't figure out there's no simple API to do that :)